src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/MailingSleepIntervalProvider.java

```diff
@@ -0,0 +1,58 @@
+package pl.tomaszdziurko.jvm_bloggers.mailing;
+
+
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import static pl.tomaszdziurko.jvm_bloggers.ApplicationProfiles.PRODUCTION;
+import static pl.tomaszdziurko.jvm_bloggers.ApplicationProfiles.STAGE;
+
+@Component
+@Slf4j
+public class MailingSleepIntervalProvider {
+
+    public static final int BASE_DELAY_IN_SECONDS = 120;
+
+    private final List<String> activeProfiles;
+
+    @Autowired
+    public MailingSleepIntervalProvider(Environment environment) {
+        activeProfiles = Arrays.asList(environment.getActiveProfiles());
+    }
+
+    public SleepInterval getSleepingInterval() {
+        if (profileIsProductionLike(activeProfiles)) {
```

Comment:
Isn't it better to use `@Profile` instead of `if` flag (in Polish "_wajcha_")? This class can be logically split into two separate implementations of a common interface.